website/docs/docs/build/cumulative-metrics.md

```diff
@@ -50,33 +57,117 @@ Cumulative metrics measure data over a given window and consider the window infi
 
 metrics:
   - name: cumulative_order_total
-    label: Cumulative Order total (All-Time)    
+    label: Cumulative order total (All-Time)    
     description: The cumulative value of all orders
     type: cumulative
     type_params:
       measure: 
         name: order_total
-        fill_nulls_with: 0
+  
   - name: cumulative_order_total_l1m
-    label: Cumulative Order total (L1M)   
-    description: Trailing 1-month cumulative order amount
+    label: Cumulative order total (L1M)   
+    description: Trailing 1-month cumulative order total
     type: cumulative
     type_params:
       measure: 
         name: order_total
-        fill_nulls_with: 0
-      window: 1 month
+      cumulative_type_params
+        window: 1 month
+  
   - name: cumulative_order_total_mtd
-    label: Cumulative Order total (MTD)
+    label: Cumulative order total (MTD)
     description: The month-to-date value of all orders
     type: cumulative
     type_params:
       measure: 
         name: order_total
-        fill_nulls_with: 0
-      grain_to_date: month
+      cumulative_type_params
+        grain_to_date: month
 ```
 
+### Granularity options
+Granularity options for cumulative metrics are slightly diffrent that granulairty for other metric types. Granularity for other metrics is implemented using the `date_trunc` function, however cumulative values are not additive so we can't simply use the `date_trunc` function to aggregate cumulative metrics.
+
+Instead we the first_value(), last_value(), and avg() aggregation functions to aggregate cumulative metrics over the requested period. By default we take the first value of the period. You can change this behaviour using the `period_agg` parameter. 
+
+Let's walk through an example using the following configs:
+
+
+```yaml
+- name: cumulative_revenue
+  description: The cumulative revenue for all orders.
+  label: Cumulative Revenue (All Time)
+  type: cumulative
+  type_params:
+    measure: revenue
+    cumulative_type_params:
+      period_agg: first # Optional. Defaults to first. Accepted values: first|end|avg
+```
+
+`period_agg` is set to first so we will choose the first value for the select granularity window. Let's say we cumulative_revenue by week using the following query: `dbt sl query --metrics cumulative_revenue --group-by metric_time__week`.
+
+This compiles the following SQl, Note the use of the window function to select the first value:
+
```

Comment:
@Jstein77 i'm not sure i understand the diff btw `first_value()`, etc. here compared to `first`. but is this placed correctly

Reply:
It's because we use the `first_value()` function in the generated SQL. We can remove this if it's confusing.